### DISCUSSION TITLE
How do I run code in VS Code visual editor mode?

### DISCUSSION THREAD

**Discussion by malcolmbarrett** [quarto-dev/quarto-cli#8807] [Q&A] [answered]
### Description

I've been working on a VS Code-first version of my Quarto teaching materials for Python. It's great, but I've had a big hangup: I can't figure out how to run code interactively *from the cell* when in visual editor mode. Is this possible? There isn't a button above the cell like in source edit mode, and the keyboard shortcut for running it doesn't work, either.

<img width="528" alt="Screenshot 2024-02-21 at 8 41 37 AM" src="https://github.com/quarto-dev/quarto-cli/assets/23123711/2c66cbf3-5c39-45a3-96f6-377f82f19aac">

**Comment by mcanouil** [comment, marked as answer]
Currently, it seems button is missing for Python.
<img width="1052" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/418a722e-973d-4b57-9775-6542b18074ec">

I suggest you open an issue at <https://github.com/quarto-dev/quarto/issues>.

**Reply by malcolmbarrett** [reply]
I didn't realize there was a different repo for that, thank you! Also helpful to see what it should look like as in the R case

**Reply by mcanouil** [reply]
The repository linked is for everything that relates to the Visual Editor which is shared across RStudio and Quarto VSCode extension.

**Reply by mcanouil** [reply]
FYI:

- https://github.com/quarto-dev/quarto/issues/103

**Reply by mcanouil** [reply]
> Currently, it is possible to execute Python cells individually in the visual editor in Positron, but not in VS Code.

https://github.com/quarto-dev/quarto/issues/103#issuecomment-2211062825

Positron: https://github.com/posit-dev/positron

**Comment by madelochan** [comment]
How to Use the Code Editor
Select the "Run Code" button to execute the program.
Select the "Calculate Grade" button to generate a score based on the completed tasks.
Continue to modify, run, and calculate your code until you are happy with the grade.
Select the "Submit" button to turn in the assignment to your instructor.
How to Use the GUI Preview
Select the "Open GUI" option from the sidebar. This will open a new tab connecting to the VNC Viewer.
Click "connect".
Enter the password: vscode

Your Tasks
The greatest common divisor of two positive integers, A and B, is the largest number that can be evenly divided into both of them. Euclid’s algorithm can be used to find the greatest common divisor (GCD) of two positive integers. You can implement this algorithm in the following manner:

Compute the remainder of dividing the larger number by the smaller number.
Replace the larger number with the smaller number and the smaller number with the remainder.
Repeat this process until the smaller number is zero.
The larger number at this point is the GCD of A and B.
Write a program in the file gcd.py that lets the user enter two integers and then prints each step in the process of using the Euclidean algorithm to find their GCD. (LO: 3.4)

**Reply by mcanouil** [reply]
Do not use IA (it seems very likely) to provide off topic answers!
